hoot-core/src/main/cpp/hoot/core/io/OsmApiCapabilites.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+#ifndef OSM_API_CAPABILITIES_H
+#define OSM_API_CAPABILITIES_H
+#include <QString>
+
+namespace hoot
+{
+//  OSM API status
+enum OsmApiStatus
+{
+  ONLINE,
+  READONLY,
+  OFFLINE
+};
+
+/**
+ *  Class to match the capabilities response from the OSM API '/api/capabilities/' request
+ *  See:    https://wiki.openstreetmap.org/wiki/API_v0.6#Capabilities:_GET_.2Fapi.2Fcapabilities
+ */
+class OsmApiCapabilites
+{
+public:
+  /**
+   *
+   */
+  OsmApiCapabilites()
+    : _version("0.6"),
+      _tracepoints(5000),
+      _waynodes(2000),
+      _changesets(10000),
+      _timeout(300),
+      _database(OsmApiStatus::OFFLINE),
+      _api(OsmApiStatus::OFFLINE),
+      _gpx(OsmApiStatus::OFFLINE)
+  {}
+
+  /**
+   *  Setter/getter for OSM API version, should be "0.6"
+   */
+  QString getVersion()                    { return _version; }
+  void setVersion(const QString& version) { _version = version; }
+
+  /**
+   *  Setter/getter for max number of GPS trace points allowed to ingest
+   */
+  long getTracepoints()                 { return _tracepoints; }
+  void setTracepoints(long tracepoints) { _tracepoints = tracepoints; }
+
+  /**
+   *  Setter/getter for max number of nodes per way allowed
+   */
+  long getWayNodes()              { return _waynodes; }
+  void setWayNodes(long waynodes) { _waynodes = waynodes; }
+
+  /**
+   *  Setter/getter for max number of elements per changeset
+   */
+  long getChangesets()                { return _changesets; }
+  void setChangesets(long changesets) { _changesets = changesets; }
+
+  /**
+   *  Setter/getter for API timeout
```

Comment:
Specify units here! Seconds? Milliseconds?

Reply:
Good call.  I'd better search the API documentation!